webapp/src/main/java/rocks/metaldetector/service/user/UserServiceImpl.java

```diff
@@ -131,11 +131,11 @@ public UserDto updateUser(String publicId, UserDto userDto) {
   @Override
   @Transactional
   public UserDto updateCurrentEmail(String emailAddress) {
-    if (userRepository.existsByEmail(emailAddress)) {
-      throw new IllegalArgumentException("emailAddress already in use");
+    UserEntity currentUser = currentUserSupplier.get();
+    if (!currentUser.getEmail().equalsIgnoreCase(emailAddress) && userRepository.existsByEmail(emailAddress)) {
```

Comment:
```suggestion
    if (currentUser.getEmail().equalsIgnoreCase(emailAddress) || userRepository.existsByEmail(emailAddress)) {
```
I find this really confusing to read here. My suggestion has a little different meaning but makes kind of more sense for me 🤔 Changing the email address to the currently existing one does not really make sense, so why allow it?